### ISSUE TITLE
When does winter begin?

### ISSUE THREAD

**Issue by JimBacon** [Indicia-Team/warehouse#150]
Winter runs from the start of December to the end of February as far as Indicia is concerned.
However, does Winter 2010 begin in 2010 and end in 2011 or does it begin in 2009 and end in 2010?

There is a bug such that 
vague_date_to_string(string_to_vague_date('Winter 2010')) = 'Winter 2011'

**Comment by JimBacon** [comment]
Our warehouse currently has no winter-year records but it might be the case that it will receive such records from Recorder 6 in future. We should aim for consistency with Recorder. John, are you able to say how Recorder behaves?

The only reference I have found to others tackling this problem is implicit in http://www.loc.gov/standards/datetime/pre-submission.html#sort where e.g. winter 2011 is sorted to follow autumn 2011 rather than precede spring 2011.

To remove some of this ambiguity I suggest that we start to output a string of e.g. Winter 2011-12 to a vague date that starts in 2011 and ends in 2012.

**Comment by johnvanbreda** [comment]
Confirming Recorder 6's behaviour - Winter 2011 starts on 2010-12-01 and ends on 2011-02-28. This originated in Recorder 3 I believe. So we are doing the opposite of the standards you have found (but to be fair, Recorder 3.3 probably preceded these standards by over 2 decades!)

**Comment by JimBacon** [comment]
I've fixed the bug but not changed the way we output the display of winter dates as I felt input and output should be matching.

**Comment by JimBacon** [comment]
The above [fix](https://github.com/Indicia-Team/warehouse/commit/4031f7fe08dbb4c867ce3daa4b6b16adefff7c4a)  addressed an inconsistency within the libraries/DateParser.php and the helpers/vague_date.php.

It omitted to update the database function, [vague_date_to_string](https://github.com/Indicia-Team/warehouse/blob/master/modules/indicia_setup/db/version_0_9_0/201308051661_vague_date_to_string.sql), which continues to consider winter as the last season of the year rather than the first.

**Comment by JimBacon** [comment]
I've checked https://warehouse1.indicia.org.uk and there are 9 winter records.
I have a supposition that users are not interpreting the winter dates as we have created them.
 - The records are all made in the spring
 - They all create records made not in the preceding winter but the one before (or two before in one case)

I envisage this as a likely scenario. In, e.g., spring 2020 I remember a sighting I made in the preceding winter and I record that as winter 2019 (mistakenly). The record is created with the date range 1/12/2018 to 28/2/2019. 

I can't know if this is the case (unless we ask the two users that have done this) but I think it is wide open to confusion at the moment. Thankfully it is a rarely used date format.
